src/OpenTelemetry/Metrics/InstrumentState.cs

```diff
@@ -0,0 +1,39 @@
+// <copyright file="InstrumentState.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+
+namespace OpenTelemetry.Metrics
+{
+    internal class InstrumentState
+    {
+        private readonly AggregatorStore store;
+
+        internal InstrumentState(MeterProviderSdk sdk, Instrument instrument)
+        {
+            this.store = new AggregatorStore(sdk, instrument);
```

Comment:
something to comeback to later - given the fact that we create a AggregatorStore *per* instrument, we need to further evaluate if there is actual perf benefit of using the 2-step-looking of aggregator for each combination of tags.

Reply:
agreed.